File: backend/E-Learning.DAL/Models/Rating.cs
using System.ComponentModel.DataAnnotations;

namespace E_Learning.DAL.Models
{
    public class Rating
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public int CourseId { get; set; }
        public Course Course { get; set; } = null!;
        public int Value { get; set; }
        [MaxLength(500)]
        public string? Description { get; set; } 
    }
}

File: frontend/src/app/models/Rating.ts
export interface Rating {
  id: number;
  userId: number;
  courseId: number;
  value: number;
  description?: string;
}

export interface RatingDTO {
  courseId: number;
  value: number;
  description?: string;
}

export interface ReadCourseRatingDTO {
  id: number;
  description: string;
  value: number;
  student: {
    id: number;
    profilePicture?: string;
    name: string;
  };
}

export interface ReadCourseRatingByUserDTO {
  id: number;
  description: string;
  value: number;
  course: {
    id: number;
    title: string;
    rate: number;
  };
}
